[SYCL][BLAS] Move portBLAS to UXL Foundation as generic SYCL BLAS (#618)
oneapi-src/oneMKL: docs/building_the_project_with_dpcpp.rst

Backends should be enabled by setting ``-DENABLE_<BACKEND_NAME>_BACKEND=True``
for each desired backend. By default, only the ``MKLGPU`` and ``MKLCPU``
backends are enabled. Multiple backends for multiple device vendors can be

> enabled at once (albeit with limitations when using onemath_sycl_blas and portFFT). The

Reviewer: ```suggestion
enabled at once (albeit with limitations when using oneMath SYCL BLAS and portFFT). The
```

> enabled at once (albeit with limitations when using oneMath SYCL BLAS and portFFT). The

Reviewer 2: Should this be "generic SYCL BLAS" or "oneMath generic SYCL BLAS"?

Consequently, it is possible to use oneMath with accelerators not anticipated by
the project.

For generic SYCL devices, only generic BLAS and portFFT backend are enabled.
The user must set the appropriate ``-fsycl-targets`` for their device, and also
any other option required for performance. See `Building for oneMath generic SYCL BLAS`_ and
`Building for portFFT`_. Extensive testing is strongly advised for these
unsupported configurations.

.. _build_for_portlibs_dpcpp:

Pure SYCL backends: generic BLAS and portFFT

> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Reviewer: ```suggestion
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

It's missing 2 to underline the whole title

> `Generic BLAS <https://github.com/codeplaysoftware/portBLAS>`_ and `portFFT

Reviewer 2: ```suggestion
`Generic SYCL BLAS <https://github.com/codeplaysoftware/portBLAS>`_ and `portFFT
```

For best performance, both libraries must be tuned. See the individual sections
for more details.

> Both onemath_sycl_blas and portFFT are used as header-only libraries, and will be

Reviewer: ```suggestion
Both oneMath SYCL BLAS and portFFT are used as header-only libraries, and will be
```

> Both oneMath SYCL BLAS and portFFT are used as header-only libraries, and will be

Reviewer 2: ```suggestion
Both generic SYCL BLAS and portFFT are used as header-only libraries, and will be
```

downloaded automatically if not found.

.. _build_for_generic_blas_dpcpp:

Building for oneMath generic SYCL BLAS

> ---------------------

Reviewer: ```suggestion
Building for oneMath generic SYCL BLAS
--------------------------------------
```

`onemath_sycl_blas generic BLAS <https://github.com/codeplaysoftware/portBLAS>`_ is
enabled by setting ``-DENABLE_GENERIC_BLAS_BACKEND=True``.

By default, the generic BLAS backend is not tuned for any specific device.
This tuning is required to achieve best performance.

> OneMath_sycl_blas can be tuned for a specific hardware target by adding compiler

Reviewer: ```suggestion
OneMath SYCL BLAS can be tuned for a specific hardware target by adding compiler
```
I think this is more consistent with the rest of the document?

> OneMath SYCL BLAS can be tuned for a specific hardware target by adding compiler

Reviewer 2: ```suggestion
The generic BLAS backend can be tuned for a specific hardware target by adding compiler
```

      -DCMAKE_C_COMPILER=clang \ 
      -DENABLE_MKLCPU_BACKEND=False \ 
      -DENABLE_MKLGPU_BACKEND=False \
      -DENABLE_GENERIC_BLAS_BACKEND=True

Note that this is not a tested configuration. This builds oneMath with the
generic BLAS backend only, for a generic SYCL device supported by the

> Open DPC++ project.

Reviewer 2: What is the "Open DPC++ project"? I recommend ending the sentence after "for a generic SYCL device."
